### PR TITLE
	fix(Compiler): allow "." in attribute selectors

### LINKED ISSUE #13645: Template Parse Errors
When I use Flex-Layout API:

```html
<div fxShow [fxShow.xs]="menuOpen" class="banner">
  I should hide and show
</div>  
```

All works fine. When I do not specify the default, non-responsive selector `fxShow`:

```html
<div [fxShow.xs]="menuOpen" class="banner">
  I should hide and show
</div>  
```

 then a Template Parse error occurs:

```terminal
Uncaught (in promise): Error: Template parse errors:
Can't bind to 'fxShow.xs' since it isn't a known property of 'div'. ("
    </button>
    
    <div [ERROR ->][fxShow.xs]="menuOpen" class="banner">
      I should hide and show
    </div>  
"): TestApp@5:9
Error: Template parse errors:
Can't bind to 'fxShow.xs' since it isn't a known property of 'div'. ("
    </button>
```

Here is the [Plunkr Demo](https://plnkr.co/edit/pZHpB08iaLiO865ieWPI?p=preview) that shows this.

> Note: I am using latest bundles on NPM: `https://unpkg.com/angular/core/bundles/core.umd.js`

### PR DESCRIPTION
**Pull request by vicb**
fixes #13645

### CODE REVIEW

File: modules/@angular/compiler/src/template_parser/template_parser.ts [hunk 1]
```diff
@@ -268,14 +268,17 @@ class TemplateParseVisitor implements html.Visitor {
...
-      let prefixToken: string|undefined = undefined;
-      if (this._normalizeAttributeName(attr.name) == TEMPLATE_ATTR) {
+      let templateBindingsSource: string|undefined;
+      let prefixToken: string|undefined;
+      let normalizedName = this._normalizeAttributeName(attr.name);
+
+      if (normalizedName == TEMPLATE_ATTR) {
         templateBindingsSource = attr.value;
-      } else if (attr.name.startsWith(TEMPLATE_ATTR_PREFIX)) {
+      } else if (normalizedName.startsWith(TEMPLATE_ATTR_PREFIX)) {
```
Comment (tbosch) [inline modules/@angular/compiler/src/template_parser/template_parser.ts]: Add a test?

File: modules/@angular/compiler/test/template_parser/template_parser_spec.ts [hunk 2]
```diff
@@ -623,6 +623,23 @@ Binding to attribute 'onEvent' is disallowed for security reasons ("<my-componen
              ]);
            });
 
+        it('should parse directive dotted properties', () => {
```
Comment (tbosch) [inline modules/@angular/compiler/test/template_parser/template_parser_spec.ts]: Add a full integration tests as well that also tests that we write the value into the property.

File: modules/@angular/compiler/test/selector_spec.ts [hunk 3]
```diff
@@ -404,3 +408,11 @@ export function main() {
     });
   });
 }
+
+function sel({tag = '', attrs = [], classes = ''}: {tag?: string,
+                                                    attrs?: any[],
+                                                    classes?: string} = {}): CssSelector {
+  const classAttr: [string, string] = ['class', classes];
+  attrs.push(classAttr);
+  return createElementCssSelector(tag, attrs);
```
Comment (tbosch) [inline modules/@angular/compiler/test/selector_spec.ts]: Copy simplified version of `createElementCssSelector` into this spec (e.g. without namespace handling, ...)

File: modules/@angular/compiler/src/template_parser/template_parser.ts (hunk 1 above)
Comment (vicb) [inline modules/@angular/compiler/src/template_parser/template_parser.ts]: done

File: modules/@angular/compiler/test/template_parser/template_parser_spec.ts (hunk 2 above)
Comment (vicb) [inline modules/@angular/compiler/test/template_parser/template_parser_spec.ts]: done

File: modules/@angular/compiler/test/integration_spec.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+             declarations: [
+               MyDir,
+               TestComponent,
+             ],
+           });
+
+           const template = `<div [dot.name]="'foo'"></div>`;
+           fixture = createTestComponent(template);
+           fixture.detectChanges();
+           const myDir = fixture.debugElement.query(By.directive(MyDir)).injector.get(MyDir);
```
Comment (ThomasBurleson) [inline modules/@angular/compiler/test/integration_spec.ts]: I did not know you could do this! Cool technique.

File: modules/@angular/compiler/test/integration_spec.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Component, Directive, Input} from '@angular/core';
+import {ComponentFixture, TestBed, async} from '@angular/core/testing';
+import {By} from '@angular/platform-browser/src/dom/debug/by';
+import {expect} from '@angular/platform-browser/testing/matchers';
+
+export function main() {
+  describe('integration tests', () => {
```
Comment (ThomasBurleson) [inline modules/@angular/compiler/test/integration_spec.ts]: Really glad to see this test. Makes it very clear. Thx.

File: modules/@angular/compiler/test/template_parser/template_parser_spec.ts [hunk 6]
```diff
@@ -623,6 +623,23 @@ Binding to attribute 'onEvent' is disallowed for security reasons ("<my-componen
...
+                    selector: '[dot.name]',
+                    type: createTypeMeta({reference: {filePath: someModuleUrl, name: 'DirA'}}),
+                    inputs: ['localName: dot.name'],
+                  })
+                  .toSummary();
+
+          expect(humanizeTplAst(parse('<div [dot.name]="expr"></div>', [dirA]))).toEqual([
+            [ElementAst, 'div'],
+            [DirectiveAst, dirA],
+            [BoundDirectivePropertyAst, 'localName', 'expr'],
```
Comment (ThomasBurleson) [inline modules/@angular/compiler/test/template_parser/template_parser_spec.ts]: Nice.

**Pull request merged**